mex/extractors/datenkompass/transform.py

```diff
@@ -2,34 +2,43 @@
 
 from mex.common.models import (
     MergedActivity,
+    MergedBibliographicResource,
     MergedOrganizationalUnit,
 )
 from mex.common.types import (
+    AccessRestriction,
+    Link,
     MergedOrganizationalUnitIdentifier,
+    MergedPersonIdentifier,
 )
 from mex.common.types.vocabulary import BibliographicResourceType, Theme
-from mex.extractors.datenkompass.models.item import DatenkompassActivity
+from mex.extractors.datenkompass.models.item import (
+    DatenkompassActivity,
+    DatenkompassBibliographicResource,
+)
 
 
 def get_contact(
     responsible_unit_ids: list[MergedOrganizationalUnitIdentifier],
-    merged_organizational_units: list[MergedOrganizationalUnit],
+    merged_organizational_units: dict[
+        MergedOrganizationalUnitIdentifier, MergedOrganizationalUnit
+    ],
 ) -> list[str]:
     """Get shortName and email from merged units.
 
     Args:
         responsible_unit_ids: List of responsible unit identifiers.
-        merged_organizational_units: List of merged organizational unit identifiers.
+        merged_organizational_units: dict of all merged organizational units by id.
 
     Returns:
         List of short name and email of contact units as strings.
     """
     return [
         contact
-        for target_unit in merged_organizational_units
-        if target_unit.identifier in responsible_unit_ids
-        for contact in [short_name.value for short_name in target_unit.shortName]
-        + [str(email) for email in target_unit.email]
+        for org_id in responsible_unit_ids
+        for unit in [merged_organizational_units[org_id]]
+        for contact in [short_name.value for short_name in unit.shortName]
+        + [str(email) for email in unit.email]
```

Comment:
```suggestion
        for org_id in responsible_unit_ids
        for contact in [short_name.value for short_name in merged_organizational_units[org_id].shortName]
        + [str(email) for email in merged_organizational_units[org_id].email]
```
Would remove one nesting layer

Reply:
can't sign off and commit anymore, because the code is outdated by other adaptations. But I implemented the sugesstion by hand.